### PR TITLE
Cask: automatically add cask audits

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

This puts the cask audits more in line with formula and allows us to select a single one at a time later.

### CODE REVIEW

**Review by Rylan12 — COMMENTED** [review]
Looks good once CI passes. Although `only` and `except` also need to be added to the [`Cask::Cmd::Audit.audit_casks` call in `brew audit`](https://github.com/Homebrew/brew/blob/fb2b21b2988ad658c3ecad2acc236d70303e5328/Library/Homebrew/dev-cmd/audit.rb#L222-L236) for this to work

Edit: oh, are you intentionally delaying that?

**Comment by SMillerDev** [comment]
I still have one failing test but I'm not sure how to resolve that one.

---

@Rylan12 I was intentionally keeping this PR separate from the changes needed to enable the CLI flags, yes.

File: Library/Homebrew/cask/audit.rb [hunk 1]
```diff
@@ -150,12 +137,9 @@ def summary(include_passed: false, include_warnings: true)
       summary.join("\n")
     end
 
-    def success?
-      !(errors? || warnings?)
-    end
-
-    private
+    protected
```
Comment (reitermarkus) [inline Library/Homebrew/cask/audit.rb:140]: I think this can stay `private`.

Comment (SMillerDev) [reply Library/Homebrew/cask/audit.rb:140]: Nope, `.methods` won't pick up on those

Comment (reitermarkus) [reply Library/Homebrew/cask/audit.rb:140]: `private_methods` then?

File: Library/Homebrew/cask/audit.rb [hunk 2]
```diff
@@ -802,6 +710,11 @@ def check_https_availability
                                        strict:        strict?)
     end
 
+    private
+
+    sig {
+      params(url_to_check: URL, url_type: String, cask_token: String, tap: Tap, options: T::Array[T.untyped]).void
```
Comment (reitermarkus) [inline Library/Homebrew/cask/audit.rb:716]:
```suggestion
      params(url_to_check: T.any(String, URL), url_type: String, cask_token: String, tap: Tap, options: T::Array[T.untyped]).void
```

File: Library/Homebrew/cask/audit.rb [hunk 3]
```diff
@@ -782,27 +689,32 @@ def check_reverse_migration
...
+      validate_url_for_https_availability(cask.homepage, "homepage URL", cask.token, cask.tap,
+                                          user_agents:   [:browser, :default],
+                                          check_content: true,
+                                          strict:        strict?)
     end
 
-    def check_url_for_https_availability(url_to_check, url_type, cask_token, tap, **options)
+    sig {
+      params(url_to_check: T.any(String, URL), url_type: String, cask_token: String, tap: Tap,
+             options: T::Array[T.untyped]).void
```
Comment (reitermarkus) [inline Library/Homebrew/cask/audit.rb:715]:
```suggestion
             options: T::Hash[String, T.untyped]).void
```

File: Library/Homebrew/cask/audit.rb [hunk 4]
```diff
@@ -782,27 +689,32 @@ def check_reverse_migration
...
+      validate_url_for_https_availability(cask.homepage, "homepage URL", cask.token, cask.tap,
+                                          user_agents:   [:browser, :default],
+                                          check_content: true,
+                                          strict:        strict?)
     end
 
-    def check_url_for_https_availability(url_to_check, url_type, cask_token, tap, **options)
+    sig {
+      params(url_to_check: T.any(String, URL), url_type: String, cask_token: String, tap: Tap,
+             options: T::Hash[String, T.untyped]).void
```
Comment (reitermarkus) [inline Library/Homebrew/cask/audit.rb:715]:
```suggestion
             options: T::Hash[Symbol, T.untyped]).void
```

**Comment by SMillerDev** [comment]
Okay, all tests fixed.

File: Library/Homebrew/cask/audit.rb [hunk 5]
```diff
@@ -47,44 +47,22 @@ def initialize(cask, appcast: nil, download: nil, quarantine: nil,
...
-      check_gitlab_repository
-      check_gitlab_repository_archived
-      check_gitlab_prerelease_version
-      check_github_repository
-      check_github_repository_archived
-      check_github_prerelease_version
-      check_bitbucket_repository
-      check_signing
+      only_audits = @only
+      except_audits = @except
```
Comment (reitermarkus) [inline Library/Homebrew/cask/audit.rb:56]: I think we should check somewhere if these are correct and actually exist. Currently we are only skipping if they match. If there's a typo it is ignored silently.

Comment (SMillerDev) [reply Library/Homebrew/cask/audit.rb:56]: We don't do that for formulae either. I'm fine adding it for both but that should be a new PR.

**Review by Rylan12 — APPROVED** [review]
Looks good. It looks like some of the methods that were re-ordered can be put back in their original spots now that the `private`/`protected` changes have been removed. I don't think it's necessary, but might be nice for diff purposes

**Review by reitermarkus — APPROVED** [review]

**Pull request merged**